### PR TITLE
Adding details to Conformance Tests using RFC 2119 standards.

### PR DESCRIPTION
**Pull request by brahmaroutu**
This PR is part of the conformance documentation. This is to provide more formal specification using RFC 2119 keywords to describe the test so that who ever is running conformance tests do not have to go through the code to understand why and what is tested.
The documentation information added here into each of the tests eventually result into a document which is currently checked in at location https://github.com/cncf/k8s-conformance/blob/master/docs/KubeConformance-1.9.md

I would like to have this PR reviewed for v1.10 as I consider it important to strengthen the conformance documents.

### CODE REVIEW

**Comment by brahmaroutu** [comment]
@WilliamDenniss @duglin @bradtopol Please review conformance documentation.

**Comment by dims** [comment]
/uncc @dims 

cc @kubernetes/sig-node-pr-reviews

**Comment by brahmaroutu** [comment]
@derekwaynecarr Can you please review the PR, it only affects comments.

File: test/e2e/node/pods.go [hunk 1]
```diff
@@ -47,6 +47,11 @@ var _ = SIGDescribe("Pods Extended", func() {
 			podClient = f.PodClient()
 		})
 		// Flaky issue #36821.
+		/*
+			Release : v1.9
+			Testname: Pods, delete grace period
+			Description: Create a pod, make sure it is running, create a watch to observe Pod creation. Create a 'kubectl local proxy', capture the port the proxy is listening. Using the http client send a ‘delete’ with gracePeriodSeconds=30. Pod SHOULD get deleted within 30 seconds.
```
Comment (spiffxp) [inline test/e2e/node/pods.go]: should the SHOULD be a MUST?

**Pull request merged**